modules/custom/openy_search/openy_google_search/openy_google_search.module

```diff
@@ -29,3 +29,11 @@ function openy_google_search_preprocess_menu__account(&$variables) {
   // Enable search form in account menu when module enabled.
   $variables['display_search'] = TRUE;
 }
+
+/**
+ * Implements hook_preprocess_menu().
+ */
+function openy_google_search_preprocess_menu__main(&$variables) {
+  // Enable search form in main menu when module enabled.
+  $variables['display_search'] = TRUE;
```

Comment:
Still, I don't like this. Instead of this, I would set the current theme setting when the search module is being enabled/disabled